src/unix/ibmi.c

```diff
@@ -55,19 +55,146 @@
 #include <strings.h>
 #include <sys/vnode.h>
 
+#include "as400_protos.h"
+
+
+typedef struct {
+  int bytes_available;
+  int bytes_returned;
+  char current_date_and_time[8];
+  char system_name[8];
+  char elapsed_time[6];
+  char restricted_state_flag;
+  char reserved;
+  int percent_processing_unit_used;
+  int jobs_in_system;
+  int percent_permanent_addresses;
+  int percent_temporary_addresses;
+  int system_asp;
+  int percent_system_asp_used;
+  int total_auxiliary_storage;
+  int current_unprotected_storage_used;
+  int maximum_unprotected_storage_used;
+  int percent_db_capability;
+  int main_storage_size;
+  int number_of_partitions;
+  int partition_identifier;
+  int reserved1;
+  int current_processing_capacity;
+  char processor_sharing_attribute;
+  char reserved2[3];
+  int number_of_processors;
+  int active_jobs_in_system;
+  int active_threads_in_system;
+  int maximum_jobs_in_system;
+  int percent_temporary_256mb_segments_used;
+  int percent_temporary_4gb_segments_used;
+  int percent_permanent_256mb_segments_used;
+  int percent_permanent_4gb_segments_used;
+  int percent_current_interactive_performance;
+  int percent_uncapped_cpu_capacity_used;
+  int percent_shared_processor_pool_used;
+  long main_storage_size_long;
+} SSTS0200;
+
+
+static int get_ibmi_system_status(SSTS0200* rcvr) {
+  // rcvrlen is input parameter 2 to QWCRSSTS
+  unsigned int rcvrlen = sizeof(*rcvr);
+  
+  // format is input parameter 3 to QWCRSSTS ("SSTS0200" in EBCDIC)
+  unsigned char format[] = {0xE2, 0xE2, 0xE3, 0xE2, 0xF0, 0xF2, 0xF0, 0xF0};
+  
+  // reset_status is input parameter 4 to QWCRSSTS ("*NO       " in EBCDIC)
+  unsigned char reset_status[] = {0x5C, 0xD5, 0xD6, 0x40, 0x40, 0x40, 0x40, 0x40, 0x40, 0x40}; 
+
+  // errcode is input parameter 5 to QWCRSSTS
+  struct _errcode {
+    int bytes_provided;
+    int bytes_available;
+    char msgid[7];
+  } errcode;
+  
+  // qwcrssts_pointer will contain the IBM i 16-byte tagged system pointer to QWCRSSTS
+  ILEpointer __attribute__((aligned(16))) qwcrssts_pointer;
+
+  // qwcrssts_argv is the array of argument pointers to QWCRSSTS
+  void *qwcrssts_argv[6];
+
+  // Set the IBM i pointer to the QSYS/QWCRSSTS *PGM object
+  int rc = _RSLOBJ2(&qwcrssts_pointer, RSLOBJ_TS_PGM, "QWCRSSTS", "QSYS");
+  
+  if(rc != 0) {
```

Comment:
Style: space between `if` and `(`. It'd be more idiomatically libuv to write this braceless:
```c
if (rc != 0)
  return rc;
```
Also applies to the code on lines 156, 175, etc.